lib/inventory_refresh/save_collection/saver/concurrent_safe_batch.rb

```diff
@@ -217,6 +217,8 @@ def load_and_update_records!(records_batch_iterator, inventory_objects_index, at
       def changed?(record)
         return true unless inventory_collection.check_changed?
 
+        # If object was archived before, pass it to update so it can be unarchived
+        return true if record.respond_to?(:archived_at) && record.archived_at
```

Comment:
@wouldn't the archived_at column be part of the changed_attributes?

Reply:
@agrare only if we'd send it explicitly, which in normal cases we won't

Reply:
Okay maybe we can set that if it not archived by default instead?  I'd rather not encode manageiq or topo specifics here if we can avoid it.

Reply:
Okay maybe we can set that if it not archived by default instead? I'd rather not encode manageiq or topo specifics here if we can avoid it.

Reply:
Ah, so usage of `:archived_at` is already in the code, it's not necessarily Topo specific, it's a required column for archiving. 

We just haven't document all the internal columns InventoryRefresh needs. Also if we'll need it, we can extract this to be configurable. :-)